File: dbflow-tests/src/test/java/com/raizlabs/android/dbflow/test/structure/ManyToManyModel.java
package com.raizlabs.android.dbflow.test.structure;

import com.raizlabs.android.dbflow.annotation.Column;
import com.raizlabs.android.dbflow.annotation.ManyToMany;
import com.raizlabs.android.dbflow.annotation.PrimaryKey;
import com.raizlabs.android.dbflow.annotation.Table;
import com.raizlabs.android.dbflow.structure.BaseModel;
import com.raizlabs.android.dbflow.test.TestDatabase;

/**
 * Description: Tests code gen on many to many.
 */
@Table(database = TestDatabase.class)
@ManyToMany(referencedTable = TestModel1.class)
public class ManyToManyModel extends BaseModel {

    @PrimaryKey
    String name;

    @PrimaryKey
    int id;

    @Column
    char anotherColumn;
}

File: dbflow-tests/src/test/java/com/raizlabs/android/dbflow/structure/ManyToManyModel.kt
package com.raizlabs.android.dbflow.structure

import com.raizlabs.android.dbflow.TestDatabase
import com.raizlabs.android.dbflow.annotation.*

/**
 * Description: Tests code gen on many to many.
 */
@Table(database = TestDatabase::class)
@ManyToMany(referencedTable = TestModel1::class)
@MultipleManyToMany(ManyToMany(referencedTable = TestModel2::class), ManyToMany(referencedTable = TestModel3::class))
class ManyToManyModel : BaseModel() {

    @PrimaryKey
    var name: String? = null

    @PrimaryKey
    var id: Int = 0

    @Column
    var anotherColumn: Char = ' '
}
